conda_forge_tick/cli.py

```diff
@@ -43,98 +43,120 @@ def deploy(args):
 
     # TODO: have function construct this
     BUILD_URL = os.environ.get(BUILD_URL_KEY, "")
-    cmds = ["git pull -s recursive -X theirs"]
-    cmds += [
-        "git add " + v
-        for v in [
-            "pr_json/*",
-            "status/*",
-            "node_attrs/*",
-            "audits/*",
-            "audits/grayskull/*",
-            "audits/depfinder/*",
-            "versions/*",
-            "profiler/*",
-            "mappings/*",
-            "mappings/pypi/*",
-        ]
-    ]
-    cmds += [f'git commit -am "Update Graph {BUILD_URL}"']
-
-    for cmd in cmds:
+
+    # pull changes, add ours, make a commit
+    try:
+        _run_git_cmd("git pull -s recursive -X theirs")
+    except Exception as e:
+        print(e)
+
+    files_to_add = (
+        # untracked
+        subprocess.run(
+            "git ls-files -o --exclude-standard",
+            shell=True,
+            capture_output=True,
+        ).stdout.decode("utf-8").splitlines()
+        # changed
+        + subprocess.run(
+            "git diff --name-only",
+            shell=True,
+            capture_output=True,
+        ).stdout.decode("utf-8").splitlines()
+        # modified and staged but not deleted
+        + subprocess.run(
+            "git diff --name-only --cached --diff-filter=d",
+            shell=True,
+            capture_output=True,
+        ).stdout.decode("utf-8").splitlines()
+    )
+    n_added = 0
+    for file in files_to_add:
+        if file is not None and len(file) > 0 and os.path.exists(file):
+            try:
+                print(f"committing: {file}", flush=True)
+                _run_git_cmd(f"git add {file}")
+                n_added += 1
+            except Exception as e:
+                print(e)
+
+    if n_added > 0:
         try:
-            _run_git_cmd(cmd)
+            _run_git_cmd(f'git commit -am "Update Graph {BUILD_URL}"')
         except Exception as e:
             print(e)
 
-    try:
-        gx = load_graph()
-        # TODO: be more selective about which json to check
-        for node, attrs in gx.nodes.items():
-            attrs["payload"]._load()
-        graph_ok = True
-    except Exception:
-        graph_ok = False
-
-    status = 1
-    num_try = 0
-    while status != 0 and num_try < 10 and graph_ok:
+        # make sure the graph can load, if not we will error
         try:
-            if num_try == 9:
+            gx = load_graph()
+            # TODO: be more selective about which json to check
+            for node, attrs in gx.nodes.items():
```

Comment:
Do you want to use your file list here to check only the nodes that have been updated? (If any)

Reply:
Not worth the effort.